code/modules/mob/living/simple_animal/hostile/gorilla/gorilla.dm

```diff
@@ -64,8 +74,23 @@
 	if(. && isliving(target))
 		var/mob/living/L = target
 		if(prob(80))
-			var/atom/throw_target = get_edge_target_turf(L, dir)
-			L.throw_at(throw_target, rand(1,2), 7, src) 
+			var/targetviewers = list()
+			var/atom/throw_target
+			var/targettingviewer
+			//the following code attempts to find a non-jungle mob within sight range of the gorilla
+			//if it finds one or more, it picks one to throw its current attack target at
+			for(var/mob/living/M in oviewers(7, src))
+				if(!("jungle" in M.faction) && M != L)
```

Comment:
this should be a proper faction check, maybe use the target list proc minus the attacked target?

Reply:
how do i do a 'proper faction check'